tests/test_computeNextBump.py

```diff
@@ -0,0 +1,259 @@
+                {
+                    'x':      1.00,
+                    'y':      1.00,
+                    'width':  1.00,
+                    'height': 1.00,
+                }
+            ]
+        },
+
+        # (bumpX, bumpY, bumpTime)
+        'out': (None, None, None)
+    },
+
+    # horizontally towards obstacle
+    {
+        'in': {
+            'currentX': 0.00,
+            'currentY': 1.50,
+            'heading':  90.00,
+            'speed':    1.00,
+            'obstacles': [
+                {
+                    'x':      1.00,
+                    'y':      1.00,
+                    'width':  1.00,
+                    'height': 1.00,
+                }
+            ]
+        },
+
+        # (bumpX, bumpY, bumpTime)
+        'out': (1.00, 1.50, 1.00)
+    },
+
+    # current position on obstacle and moving towards obstacle
+    {
+        'in': {
+            'currentX': 1.00,
+            'currentY': 1.50,
+            'heading':  90.00,
+            'speed':    1.00,
+            'obstacles': [
+                {
+                    'x': 1.00,
+                    'y': 1.00,
+                    'width': 1.00,
+                    'height': 1.00,
+                }
+            ]
+        },
+
+        # (bumpX, bumpY, bumpTime)
+        'out': (1.00, 1.50, 0)
+    },
+
+    # horizontally away from obstacle
+    {
+        'in': {
+            'currentX': 2.00,
+            'currentY': 1.50,
+            'heading':  90.00,
+            'speed':    1.00,
+            'obstacles': [
+                {
+                    'x':      1.00,
+                    'y':      1.00,
+                    'width':  1.00,
+                    'height': 1.00,
+                }
+            ]
+        },
+
+        # (bumpX, bumpY, bumpTime)
+        'out': (None, None, None)
+    },
+
+    # horizontally towards 2 obstacles
+    {
+        'in': {
+            'currentX': 0.00,
+            'currentY': 1.50,
+            'heading':  90.00,
+            'speed':    1.00,
+            'obstacles': [
+                {
+                    'x':      1.00,
+                    'y':      1.00,
+                    'width':  1.00,
+                    'height': 1.00,
+                },
+                {
+                    'x':      1.00,
+                    'y':      3.00,
+                    'width':  1.00,
+                    'height': 1.00,
+                }
+
+            ]
+        },
+
+        # (bumpX, bumpY, bumpTime)
+        'out': (1.00, 1.50, 1.00)
+    },
+
+]
+
+@pytest.fixture(params=EXPECTEDINOUT)
+def expectedInOut(request):
+    return request.param
+
+# ============================ tests =================================
+
+def test_computeNextBump(expectedInOut):
+    '''
+    testing computation of next bump with obstacle coordinates and time
+    '''
+
+    floorplan = '''
+        ######
+        #...s#
+        ######
+    '''
+    dotBot = DotBot.DotBot(1,0,0, floorplan)
```

Comment:
put in the name of the parameters when intanstation DotBot. Indent nicely, one param per line